src/main/java/duke/helpers/Parser.java

```diff
@@ -0,0 +1,97 @@
+package duke.helpers;
+
+import duke.exceptions.DukeException;
+import duke.exceptions.EmptyDescriptionException;
+import duke.exceptions.EmptyTimeException;
+import duke.exceptions.InvalidCommandException;
+import duke.commands.Command;
+
+import java.util.regex.Pattern;
+
+/**
+ * Represents a parser which makes sense of user inputs and calls the calls the corresponding commands.
+ */
+public class Parser {
+
+    /**
+     * Determines the type of commands given in the user input.
+     *
+     * @param s user input.
+     * @param command type of command to be checked against.
+     * @return boolean that indicates if the user gave the specified command.
+     * @throws DukeException If deadline and event commands are missing date and/or time.
+     */
+    static boolean isCommand(String s, Command command) throws DukeException {
```

Comment:
I really like the idea of having static methods for Parser actually, then we can just use Parser like a library of static methods since we don't really need to instantiate multiple Parser objects each with their own state. I might make that change for my own implementation